packages/kit/src/runtime/server/parse_body/index.js

```diff
@@ -2,30 +2,37 @@ import { read_only_form_data } from './read_only_form_data.js';
 
 /**
  * @param {import('types/hooks').StrictBody} raw
- * @param {import('types/helper').Headers} headers
+ * @param {Partial<import('types/helper').Headers>} headers
  */
 export function parse_body(raw, headers) {
-	if (!raw || typeof raw !== 'string') return raw;
+	if (!raw) return raw;
 
-	const [type, ...directives] = headers['content-type'].split(/;\s*/);
+	const content_type = headers['content-type'];
+	const [type, ...directives] = content_type ? content_type.split(/;\s*/) : [];
+
+	const text = () =>
+		typeof raw === 'string'
+			? raw
+			: new TextDecoder(headers['content-encoding'] || 'utf-8').decode(raw);
+	const bytes = () => (typeof raw === 'string' ? new TextEncoder().encode(raw) : raw);
```

Comment:
assuming `raw` is always a `Uint8Array` can we simplify this?

```suggestion
	const text = () => new TextDecoder(headers['content-encoding'] || 'utf-8').decode(raw);
```